provider/dataset/dataset.go

```diff
@@ -148,9 +148,69 @@ type SizedIterator interface {
 
 type GenericSizedIterator struct {
 	Iterator
-	length int64
+	Length int64
 }
 
 func (it *GenericSizedIterator) Len() (int64, error) {
-	return it.length, nil
+	return it.Length, nil
 }
+
+/**
```

Comment:
Are you planning on leaving this commented out or is this an artifact of your work process that should be deleted?